frontend/src/pages/Metrics/index.tsx

```diff
@@ -1,91 +1,55 @@
 import { Space } from 'antd';
-import getLocalStorageKey from 'api/browser/localstorage/get';
+import getService from 'api/metrics/getService';
 import ReleaseNote from 'components/ReleaseNote';
 import Spinner from 'components/Spinner';
-import { SKIP_ONBOARDING } from 'constants/onboarding';
 import MetricTable from 'container/MetricsTable';
 import ResourceAttributesFilter from 'container/ResourceAttributesFilter';
 import { useNotifications } from 'hooks/useNotifications';
 import useResourceAttribute from 'hooks/useResourceAttribute';
 import { convertRawQueriesToTraceSelectedTags } from 'hooks/useResourceAttribute/utils';
 import { useEffect, useMemo } from 'react';
-import { connect, useSelector } from 'react-redux';
+import { useQuery } from 'react-query';
+import { useSelector } from 'react-redux';
 import { useLocation } from 'react-router-dom';
-import { bindActionCreators, Dispatch } from 'redux';
-import { ThunkDispatch } from 'redux-thunk';
-import { GetService, GetServiceProps } from 'store/actions/metrics';
 import { AppState } from 'store/reducers';
-import AppActions from 'types/actions';
 import { GlobalReducer } from 'types/reducer/globalTime';
-import MetricReducer from 'types/reducer/metrics';
 import { Tags } from 'types/reducer/trace';
 
-function Metrics({ getService }: MetricsProps): JSX.Element {
-	const { minTime, maxTime, loading, selectedTime } = useSelector<
+function Metrics(): JSX.Element {
+	const { minTime, maxTime, selectedTime } = useSelector<
 		AppState,
 		GlobalReducer
 	>((state) => state.globalTime);
-	const location = useLocation();
-	const { services, error, errorMessage } = useSelector<AppState, MetricReducer>(
-		(state) => state.metrics,
-	);
-	const { notifications } = useNotifications();
-
-	useEffect(() => {
-		if (error) {
-			notifications.error({
-				message: errorMessage,
-			});
-		}
-	}, [error, errorMessage, notifications]);
 
+	const location = useLocation();
 	const { queries } = useResourceAttribute();
 
 	const selectedTags = useMemo(
 		() => (convertRawQueriesToTraceSelectedTags(queries, '') as Tags[]) || [],
 		[queries],
 	);
 
-	const isSkipped = getLocalStorageKey(SKIP_ONBOARDING) === 'true';
-
-	useEffect(() => {
-		if (loading === false) {
+	const { data, error, isLoading } = useQuery(
+		[minTime, maxTime, selectedTime, selectedTags],
+		() =>
 			getService({
-				maxTime,
-				minTime,
+				end: maxTime,
+				start: minTime,
 				selectedTags,
-			});
-		}
-	}, [getService, loading, maxTime, minTime, selectedTags]);
+			}),
+	);
 
-	useEffect(() => {
-		let timeInterval: NodeJS.Timeout;
+	const { notifications } = useNotifications();
 
-		if (loading === false && !isSkipped && services.length === 0) {
-			timeInterval = setInterval(() => {
-				getService({
-					maxTime,
-					minTime,
-					selectedTags,
-				});
-			}, 50000);
+	useEffect(() => {
```

Comment:
can we make a hook though we have make this call every time

Reply:
Okay